Userland/Libraries/LibWeb/DOM/AbortSignal.cpp

```diff
@@ -132,4 +134,30 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<AbortSignal>> AbortSignal::abort(JS::VM& vm
     return signal;
 }
 
+// https://dom.spec.whatwg.org/#dom-abortsignal-timeout
+WebIDL::ExceptionOr<JS::NonnullGCPtr<AbortSignal>> AbortSignal::timeout(JS::VM& vm, WebIDL::UnsignedLongLong milliseconds)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Let signal be a new AbortSignal object.
+    auto signal = TRY(construct_impl(realm));
+
+    // 2. Let global be signal’s relevant global object.
+    auto& global = realm.global_object();
+    auto* window_or_worker = dynamic_cast<HTML::WindowOrWorkerGlobalScopeMixin*>(&global);
+    VERIFY(window_or_worker);
```

Comment:
```c++
auto& global = HTML::relevant_global_object(*this);
```
:)

(plus whatever casting you have to do to get to `WindowOrWorkerGlobalScopeMixin`).

Reply:
FYI to find these AOs, I just click the link from the spec text ("relevant global object" in this case) and grep the code base for the anchor (concept-relevant-global):
```bash
» cgr concept-relevant-global
Userland/Libraries/LibWeb/HTML/Scripting/Environments.cpp
415:// https://html.spec.whatwg.org/multipage/webappapis.html#concept-relevant-global
```

so then I know we can just call that. (Or if the AO hasn't been implemented yet, try to implement it to help our future selves)